### PR TITLE
[APB-3331][SM]: Cleanup Cache handling in agent-invitation-frontend

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsFastTrackInvitationController.scala [hunk 1]
```diff
@@ -173,15 +178,15 @@ class AgentsFastTrackInvitationController @Inject()(
...
           Ok(
             check_details(
               checkDetailsForm,
-              currentInvitation,
+              agentSession,
               featureFlags,
-              serviceToMessageKey(currentInvitation.service),
-              CheckDetailsPageConfig(currentInvitation, featureFlags),
+              serviceToMessageKey(agentSession.service.getOrElse("")),
+              CheckDetailsPageConfig(agentSession, featureFlags),
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsFastTrackInvitationController.scala]: Could some of these parameters for the view go into the CheckDetailsPageConfig?

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala [hunk 2]
```diff
@@ -114,55 +110,59 @@ class AgentLedDeAuthController @Inject()(
...
-            val clientName = name.getOrElse("")
-            Ok(
-              cancelAuthorisation.confirm_cancel(
-                cache.service,
+          invitationsService
+            .getClientNameByService(cache.clientIdentifier.getOrElse(""), cache.service.getOrElse(""))
+            .flatMap { name =>
+              val clientName = name.getOrElse("")
+              Ok(cancelAuthorisation.confirm_cancel(
+                cache.service.getOrElse(""),
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: Raise new ticket to decide and implement how to best deal with these missing params. Probably redirect back to most appropriate place

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsFastTrackInvitationController.scala [hunk 3]
```diff
@@ -83,14 +79,15 @@ class AgentsFastTrackInvitationController @Inject()(
 
   val showKnownFact: Action[AnyContent] = Action.async { implicit request =>
     withAuthorisedAsAgent { (_, _) =>
-      currentAuthorisationRequestCache.fetch.map {
-        case Some(currentInvitation)
-            if currentInvitation.clientIdentifier.nonEmpty && currentInvitation.clientIdentifierType.nonEmpty && currentInvitation.service.nonEmpty =>
+      agentSessionCache.fetch.map {
+        case Some(agentSession)
+            if agentSession.clientIdentifier.nonEmpty && agentSession.clientIdentifierType.nonEmpty && agentSession.service.nonEmpty =>
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsFastTrackInvitationController.scala]: Theres a couple of these complicated conditionals and case statements that could be extracted to make them easier to read. But this is minor

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsFastTrackInvitationController.scala]: yeah, if there are additional cleanups needed, then i prefer to do them in the upcoming PRs

**Pull request merged**